crates/core/tedge_api/src/entity_store.rs

```diff
@@ -1401,7 +1378,7 @@ mod tests {
                     topic_id: EntityTopicId::from_str("device/child1//").unwrap(),
                     r#type: EntityType::ChildDevice,
                     external_id: Some("device:child1".into()),
-                    parent: None,
+                    parent: Some(EntityTopicId::from_str("device/main//").unwrap()),
```

Comment:
I was surprised by this. Why the parent of child device was expected to be `None`? Because no registration is required for the main device?

Reply:
> and the parent would be resolved by the entity store when registering the child.

This is what I would like to fix with this PR. Indeed, for auto registration, the entity store starts with a topic matching the default topic scheme (starting with a validation) and then extracts sub data (type, names, relationship) using misc functions assuming each the default topic scheme.

With this PR,  the entity store can parse an entity topic id into all the info required for auto-registration is applicable.

This also addresses what [you noted](https://github.com/thin-edge/thin-edge.io/pull/3139#pullrequestreview-2328805664) : *currently it seems it has a very large surface with lots of methods that are ultimately a bit shallow*

Reply:
> So I think it'd be better to leave it as None.

However, I will opt for the other way, i.e. setting the parent. Indeed, the point of the `parse(entity_topic_id)`
is to provide an *explicit representation* of all what is implied by an entity topic of the default topic scheme.